buildSrc/src/main/kotlin/options.kt

```diff
@@ -1,20 +1,24 @@
+import com.android.build.gradle.BaseExtension
+import org.gradle.api.JavaVersion
 import org.gradle.api.Project
 import org.gradle.api.plugins.JavaPluginExtension
 import org.gradle.api.tasks.compile.JavaCompile
 import org.gradle.jvm.toolchain.JavaLanguageVersion
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmCompilerOptions
 import org.jetbrains.kotlin.gradle.dsl.KotlinJvmOptions
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
-
-fun Project.configureCompilerOptions(jvmVersion: Int = 8) {
+private fun Int.toJavaVersion(): String = when(this) {
+    8 -> "1.8"
+    else -> toString()
+}
+fun Project.configureCompilerOptions(jvmVersion: Int = 11) {
     tasks.withType(KotlinCompile::class.java).configureEach {
         it.compilerOptions {
             //allWarningsAsErrors.set(true)
-            (this as? KotlinJvmOptions)?.let {
-                it.jvmTarget = when (jvmVersion) {
-                    8 -> "1.8"
-                    else -> jvmVersion.toString()
-                }
+            (this as? KotlinJvmCompilerOptions)?.let {
```

Comment:
Is this different from?
```
   kotlinOptions {
        jvmTarget = JavaVersion.VERSION_11.toString()
    }
```

Reply:
Excellent question! I believe this is essentially the same thing. The Kotlin class hierarchy is nuts

![Screenshot 2023-03-20 at 13 38 26](https://user-images.githubusercontent.com/3974977/226341492-31c146fe-fc32-4d6e-b8bd-ed78b50b3c18.png)

I was importing `org.jetbrains.kotlin.gradle.tasks.KotlinCompile` and not `org.jetbrains.kotlin.gradle.dsl.KotlinCompile`. I changed to using the DSL one. The only important thing is to check that it is a JVM task because this might run for other targets too.